### PR TITLE
Added setmode binary for stdout on windows

### LINKED ISSUE #30: Windows requires setmode binary for stdout
I've received an issue that blocks `stdout` mode in windows: `compress -c` is not usable. The problem is legacy line feed magic coming from MS-DOS: system `write` and `0x0a` byte are the worst enemies, because `stdout` is not in binary mode by default.

We need to add `setmode(fileno(stdout), O_BINARY)` before using `stdout`.

### PR DESCRIPTION
**Pull request by andrew-aladev**
Closes #30

### CODE REVIEW

**Review by vapier — CHANGES_REQUESTED** [review]

File: compress42.c [hunk 1]
```diff
@@ -867,9 +868,17 @@ nextarg:	continue;
 			exit_code = 0;
 			remove_ofname = 0;
 
+			fdin = fileno(stdin);
+			fdout = fileno(stdout);
+
+			#if defined(WINDOWS)
+				setmode(fdin, O_BINARY);
```
[the remark this answers is not in the document]
Comment (andrew-aladev) [reply compress42.c]: Thank you, I've updated it.

**Comment by andrew-aladev** [comment]
I've used msvc with the following command `cl /O2 -o compress-windows-x64.exe -DNOFUNCDEF -DUSERMEM=800000 -DREGISTERS=3 compress42.c`. Now it works perfect like on linux and mac.

**Comment by vapier** [comment]
ah i just noticed this is against the ncompress-4.2.4 branch.  i'm not planning on making anymore releases from that.  can you redo the change against the main branch and i can look at creating the first 5.x release ?  there's more Windows fixes in the main branch that haven't been included in the ncompress-4.2.4 branch.

**Comment by andrew-aladev** [comment]
Ok, I will update pr today evening, thanks.

**Pull request closed without merging**